Silicon/CommonSocPkg/Library/GpioLib/GpioLib.inf

```diff
@@ -0,0 +1,40 @@
+## @file
+#
+#  Copyright (c) 2020, Intel Corporation. All rights reserved.<BR>
+#  SPDX-License-Identifier: BSD-2-Clause-Patent
+#
+##
+
+[Defines]
+  INF_VERSION                    = 0x00010005
+  BASE_NAME                      = GpioLib
+  FILE_GUID                      = 421461F4-5508-4a1a-8F6C-BC1034D3CBC2
+  MODULE_TYPE                    = BASE
+  VERSION_STRING                 = 1.0
+  LIBRARY_CLASS                  = GpioLib
+
+#
+# The following information is for reference only and not required by the build tools.
+#
+#  VALID_ARCHITECTURES           = IA32 X64 IPF
+#
+
+[Sources]
+  GpioLib.c
+  GpioInit.c
+  GpioHelpersLib.c

```

Comment:
For the GPIO libraries impl, from this patch it looks we only have one common GPIO library instance Silicon/CommonSocPkg/Library/GpioLib/GpioLib.inf, but it has 2 header files.  If this GpioLib.inf works for all the platforms, we should only have one library header file for this instance. (maybe merge GpioSocLib.h into GpioLib.h).

Reply:
GpioLib.h is in Core package. Not sure, if we want to put all these Gpio related APIs in Core pakcage. If it is OK, I can merge and have one single header file. Agree , it is more clean that way.

Reply:
sorry for late response. After merging GpioLib.h into current GpioSocLib.h, and then rename merged GpioSocLib.h to GpioLib.h, I think we could merge this patch.

Reply:
Following header files are there in the new patch:

GpioLib.h ---> common API for all the Platforms to call the main Gpio configuration routines. Also exposes certain functions so that platforms can use

GpioPlatformLib.h ---> Platform specific implementation of API

GpioInternalLib.h --->Prototypes for functions used within CommonSoc\Silicon\GpioLib

Reply:
thanks for the update.

Reply:
APL, CML, CMLV, CFL use the name 'GpioLib.h' for their internal GpioLib implementations. They also include CommonSocPkg.dec in their drivers. So the name is clashing with the 'Silicon/CommonSocPkg/Include/Library/GpioLib.h'. Also, the CommonSocPkg/../../GpioLib.h has some functions which are also present in these platform specific header files. So, there are build errors.

I suggest, we rename CommonSocPkg/Include/Library/GpioLib.h to GpioSocLib.h for now. And which ever platform wnats to use the common GpioLib can make changes to its Silicon/Platform code to refer to this common library functions and headers. Otherwise, it is a lot of effort to make all platforms build with this new lib at the same time.

Reply:
Agree we use GpioSocLib.h for now. once all the platform is updated to use this common lib, we could rename it to GpioLib.h.

Reply:
Thanks for the review. WIll make the chnage again. Will add EFIAPI for all interface functions in next patch.